File: ComponentLib/src/main/java/com/component/impl/IActionBarResIdProvider.java
package com.component.impl;

import android.support.annotation.IdRes;

/**
 * Created by yangweichao on 2018/3/7.
 */

public interface IActionBarResIdProvider {

    @IdRes
    int getActionBarId();

    @IdRes
    int getActionBarTitleId();

    @IdRes
    int getActionBarContainerId();

    @IdRes
    int getActionBarIconId();

    @IdRes
    int getActionBarActionId();
}

File: ComponentLib/src/main/java/com/component/widget/IActionBarResIdProvider.kt
package com.component.widget

import android.support.annotation.IdRes

/**
 * Created by alan on 2017/11/4.
 */
interface IActionBarResIdProvider {

    @IdRes
    fun getActionBarId(): Int

    @IdRes
    fun getActionBarTitleId(): Int

    @IdRes
    fun getActionBarContainerId(): Int

    @IdRes
    fun getActionBarIconId(): Int

    @IdRes
    fun getActionBarActionId(): Int
}